src/WebJobs.Script/Diagnostics/ILoggerFactoryHook.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryHook
```

Comment:
I like `Initialzer` better than `Hook`. Would be nice if we were following a "configuration builder" pattern here. But initializer taking the factory seems fine.